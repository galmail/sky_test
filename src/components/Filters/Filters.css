.filters {
    clear: both;
    overflow: hidden;
    margin-bottom: 20px;
}

.filters > div {
    float: left;
    margin: 0 30px;
}

.filters label {
    margin-right: 10px; 
}

