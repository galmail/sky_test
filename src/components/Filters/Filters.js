import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import './Filters.css';

const DATETIME_FORMAT = 'YYYY-MM-DDThh:mm';

class Filters extends Component {

  constructor(props) {
    super(props);
    this.state = {
      fromDate: props.fromDate,
      toDate: props.toDate,
    };
  }

  onChange(dateType, value) {
    const dateObj = moment(value, DATETIME_FORMAT);
    let fromDate = this.state.fromDate;
    let toDate = this.state.toDate;
    if (dateType === 'fromDate') {
      if (dateObj.isValid()) fromDate = dateObj.toDate();
      else fromDate = null;
    }
    if (dateType === 'toDate') {
      if (dateObj.isValid()) toDate = dateObj.toDate();
      else toDate = null;
    }
    this.setState({ fromDate, toDate }, () => this.props.onChange(this.state));
  }
  
  render() {
    const fromDate = this.state.fromDate ? moment(this.state.fromDate).format(DATETIME_FORMAT) : '';
    const toDate = this.state.toDate ? moment(this.state.toDate).format(DATETIME_FORMAT) : '';
    return (
      <div className="filters">
        <div className="startDate">
          <label>From:</label>
          <input id="startDateId" type="datetime-local" value={fromDate} onChange={(e) => this.onChange('fromDate', e.target.value)} />
        </div>
        <div className="endDate">
          <label>To:</label>
          <input id="endDateId" type="datetime-local" value={toDate} onChange={(e) => this.onChange('toDate', e.target.value)} />
        </div>
      </div>
    );
  }

}


Filters.propTypes = {
  fromDate: PropTypes.instanceOf(Date),
  toDate: PropTypes.instanceOf(Date),
  onChange: PropTypes.func,
};

Filters.defaultProps = {
  fromDate: new Date(1481414591537),
  toDate: new Date(1512950630150),
  onChange: () => {},
};

export default Filters;



