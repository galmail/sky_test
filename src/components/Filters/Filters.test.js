import React from 'react';
import { shallow } from 'enzyme';

import Filters from './Filters';

describe('Test Filters Component', () => {

  it('should render without crashing', () => {
    const wrapper = shallow(<Filters />);
    expect(wrapper).toMatchSnapshot();
  });

});
