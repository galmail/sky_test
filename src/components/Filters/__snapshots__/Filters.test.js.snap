exports[`Test Filters Component should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    fromDate={2016-12-11T00:03:11.537Z}
    onChange={[Function]}
    toDate={2017-12-11T00:03:50.150Z} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="startDate">
          <label>
                    From:
          </label>
          <input
                    id="startDateId"
                    onChange={[Function]}
                    type="datetime-local"
                    value="2016-12-11T12:03" />
</div>,
        <div
          className="endDate">
          <label>
                    To:
          </label>
          <input
                    id="endDateId"
                    onChange={[Function]}
                    type="datetime-local"
                    value="2017-12-11T12:03" />
</div>,
      ],
      "className": "filters",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              From:
</label>,
            <input
              id="startDateId"
              onChange={[Function]}
              type="datetime-local"
              value="2016-12-11T12:03" />,
          ],
          "className": "startDate",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From:",
            },
            "ref": null,
            "rendered": "From:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "startDateId",
              "onChange": [Function],
              "type": "datetime-local",
              "value": "2016-12-11T12:03",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              To:
</label>,
            <input
              id="endDateId"
              onChange={[Function]}
              type="datetime-local"
              value="2017-12-11T12:03" />,
          ],
          "className": "endDate",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To:",
            },
            "ref": null,
            "rendered": "To:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "endDateId",
              "onChange": [Function],
              "type": "datetime-local",
              "value": "2017-12-11T12:03",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="startDate">
            <label>
                        From:
            </label>
            <input
                        id="startDateId"
                        onChange={[Function]}
                        type="datetime-local"
                        value="2016-12-11T12:03" />
</div>,
          <div
            className="endDate">
            <label>
                        To:
            </label>
            <input
                        id="endDateId"
                        onChange={[Function]}
                        type="datetime-local"
                        value="2017-12-11T12:03" />
</div>,
        ],
        "className": "filters",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                From:
</label>,
              <input
                id="startDateId"
                onChange={[Function]}
                type="datetime-local"
                value="2016-12-11T12:03" />,
            ],
            "className": "startDate",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From:",
              },
              "ref": null,
              "rendered": "From:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "startDateId",
                "onChange": [Function],
                "type": "datetime-local",
                "value": "2016-12-11T12:03",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                To:
</label>,
              <input
                id="endDateId"
                onChange={[Function]}
                type="datetime-local"
                value="2017-12-11T12:03" />,
            ],
            "className": "endDate",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To:",
              },
              "ref": null,
              "rendered": "To:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "endDateId",
                "onChange": [Function],
                "type": "datetime-local",
                "value": "2017-12-11T12:03",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
